### PR TITLE
Added support for security in payloads

### PR DESCRIPTION
**Pull request by mejt**

### CODE REVIEW

File: public/index.php [hunk 1]
```diff
@@ -57,6 +59,9 @@
 } catch (InvalidIncomingDataException $e) {
     $response['message'] = 'Data incorrect';
     http_response_code(400); // Data are incorrect
+} catch (InvalidSecretSignatureException $e) {
+    $response['message'] = 'Security signature is incorrect';
+    http_response_code(400); // Data are incorrect
```
Comment (mikron-ia) [inline public/index.php]: I would suggest using response code 401 - Unauthorised (in stead of 400), as the request is correct, just improperly authorised.

File: src/Services/ServiceFactory.php [hunk 2]
```diff
@@ -3,13 +3,25 @@
 namespace FP\Larmo\Agents\WebHookAgent\Services;
 
 use FP\Larmo\Agents\WebHookAgent\Exceptions\EventTypeNotFoundException;
+use FP\Larmo\Agents\WebHookAgent\Exceptions\InvalidSecretSignatureException;
 use FP\Larmo\Agents\WebHookAgent\Exceptions\ServiceNotFoundException;
 
 class ServiceFactory
 {
-    public static function create($serviceName, $request)
+    public static function create($serviceName, $request, $secrets = array())
```
Comment (mikron-ia) [inline src/Services/ServiceFactory.php]: We are using array() format along with [] format; I suppose we should use one of them consistently.
For sake of brevity, **I suggest using [] everywhere**, as main Larmo project does.

File: public/index.php (hunk 1 above)
Comment (zeulus) [inline public/index.php]: Go for it. Good idea?

File: src/Services/SecuritySignatureInterface.php [hunk 3]
```diff
@@ -0,0 +1,9 @@
+<?php
+
+namespace FP\Larmo\Agents\WebHookAgent\Services;
+
+interface SecuritySignatureInterface
+{
+    public function __construct($request, $secrets);
+    public function isSetCorrectSignature();
```
Comment (mikron-ia) [inline src/Services/SecuritySignatureInterface.php]: I would suggest renaming the method "isSignatureCorrect()" or "isSecuritySignatureCorrect()", as current name is a bit misleading (and sounds more like a test).

**Pull request merged**